src/metadata.rs

```diff
@@ -1376,23 +1376,28 @@ impl TargetPath {
 }
 
 /// Description of a target, used in verification.
-#[derive(Debug, Clone, PartialEq, Serialize)]
+#[derive(Debug, Clone, PartialEq)]
 pub struct TargetDescription {
     length: u64,
     hashes: HashMap<HashAlgorithm, HashValue>,
+    custom: HashMap<String, serde_json::Value>,
```

Comment:
Shouldn't this be optional? So far as I recall, it's optional in the spec even if it's always there in `go-tuf`.

Reply:
Yes, it should be optional

Reply:
I can make it explicitly optional. I this up such that we won't add `"custom"` if the `custom` map is empty, but making this field an `Option<HashMap<String, serde_json::Value>>` would be more semantically correct.

Reply:
Actually, I've changed my mind :) I can't think of a case where I'd want to do something different if some target has an empty `"custom"` map compared to that target having no `"custom"` field. The shim in this patch serializes the custom field only if it's empty, otherwise it ignores it, and deserializes to an empty map if it's not present in the metadata. Can anyone come up with a counter example where this pattern is wrong?

Reply:
Well, technically, consumers are free to ignore custom targets metadata, if they so wish, so this should be ok.

Reply:
Won't this cause a problem for verifying metadata that contains `custom: {}` ?

Reply:
I think it is an important difference to differentiate between absent, null, and empty custom. Because we're fundamentally dealing with JSON right now and those are all different in JSON land. This library should be able to support creating and signing metadata, and that's not possible if these three cases are all treated as identical.

Reply:
good catch! Yeah that would cause the signatures to be different because we aren't internally saving the raw unparsed json value. I'll switch back to `Option<HashMap<String, Value>>`.